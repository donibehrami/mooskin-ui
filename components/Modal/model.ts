import { IDivBoxComponentProps } from '../Box/model';

export interface IModalComponentProps extends IDivBoxComponentProps {
    /** whether the Modal is open or not */
    isOpen?: boolean;

    /** call close callback when esc key is pressed */
    closeOnEsc?: boolean;

    /** call close callback when the overlay is clicked */
    closeOnOverlayClick?: boolean;

    /** close callback */
    onClose?: (e: React.MouseEvent<HTMLElement>) => void;
}

export interface IModalContentComponentProps extends IDivBoxComponentProps {
    /** whether the Modal is open or not */
    isOpen?: boolean;
}

export interface IModalOverlayComponentProps extends IDivBoxComponentProps {
    /** whether the Modal is open or not */
    isOpen?: boolean;

    /** callback when clicking the Modal overlay */
    onClickOverlay?: (e: React.MouseEvent<HTMLElement>) => void;
}

export interface IModalCloseButtonComponentProps extends IDivBoxComponentProps {
    /** callback when clicking the Modal overlay */
    onClickButton?: (e: React.MouseEvent<HTMLElement>) => void;
}
