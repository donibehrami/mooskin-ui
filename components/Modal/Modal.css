@import '../_utils/globals/screens.css';

.container{
    background: rgba(0, 0, 0, 0.5);
    transition: all 0.4s;
    overflow-x: hidden;
    overflow-y: auto;
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 1040;
    display: flex;
}

.on{
    opacity: 1;
    visibility: visible;
}

.off{
    opacity: 0;
    visibility: hidden;
}

.modal {
    position: relative;
    margin: auto;
    background: #fff;
    border-radius: 8px;
    width: 60%;
    transition: 0.4s;
    display: flex;
    flex-direction: column;
    box-shadow: 0px 10px 9px -5px rgba(60,60,60,1);
    /* z-index: 5; */
}

.modalOn{
    top: 0px;
}

.modalOff{
    top: -600px;
}

.cover{
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0px;
    left: 0px;
    z-index: -1;
}

.coverOff{
    visibility: hidden;
}

.coverOn{
    visibility: visible;
}

.header {
    background: #F3F3F3;
    border-top-left-radius: 8px;
    border-top-right-radius: 8px;
    display: flex;
    position: relative;
    height: 80px;
    border-bottom: 1px solid #DDDDDD;
}

.iconWrapper {
    position: absolute;
    right: 10px;
    top: 50%;
    margin-top: -15px;
}

@media (--small){
    
    .modal{
        width: 80%;
    }

}

@media (--xSmall){

    .modal{
        width: 80%;
    }

}