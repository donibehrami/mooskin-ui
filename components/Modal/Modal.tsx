import * as React from 'react';

// Models
import { IDivBoxComponentProps } from '../Box/model';
import { IModalCloseButtonComponentProps, IModalComponentProps, IModalContentComponentProps, IModalOverlayComponentProps } from './model';

// Styled Components
import {
    StyledModal,
    StyledModalBody,
    StyledModalCloseButton,
    StyledModalContentFadeIn,
    StyledModalContentFadeOut,
    StyledModalFooter,
    StyledModalHeader,
    StyledModalOverlayFadeIn,
    StyledModalOverlayFadeOut
} from './styles';

export const Modal: React.FC<IModalComponentProps> = (props) => {

    const batchClickHandler = (e: React.MouseEvent<HTMLDivElement>, callback?: (e: React.MouseEvent<HTMLDivElement>) => void) => {
        props.onClose && props.onClose(e);
        callback && callback(e);
    };

    const recurseChildren = (children: any): any => {
        if (!children){
            return null;
        }

        return React.Children.map(children, (child, i) => {
            if (React.isValidElement<IModalCloseButtonComponentProps>(child) && child.type === ModalCloseButton){
                return React.cloneElement(child, {
                    children: recurseChildren(child.props.children),
                    key: i,
                    onClickButton: (e) => batchClickHandler(e, child.props.onClickButton)
                } as IModalCloseButtonComponentProps);
            }

            if (React.isValidElement<IModalOverlayComponentProps>(child) && child.type === ModalOverlay){
                return React.cloneElement(child, {
                    children: recurseChildren(child.props.children),
                    isOpen: child.props.isOpen ? child.props.isOpen : props.isOpen,
                    key: i,
                    onClickOverlay: props.closeOnOverlayClick ?
                                (e) => batchClickHandler(e, child.props.onClickOverlay) :
                                undefined
                } as IModalOverlayComponentProps);
            }

            if (React.isValidElement<IModalContentComponentProps>(child) && child.type === ModalContent){
                return React.cloneElement(child, {
                    children: recurseChildren(child.props.children),
                    isOpen: child.props.isOpen ? child.props.isOpen : props.isOpen,
                    key: i,
                    onClick: (e: React.MouseEvent<HTMLDivElement>) => {
                        e.stopPropagation();
                        child.props.onClick && child.props.onClick(e);
                    }
                } as IModalContentComponentProps);
            }

            if (React.isValidElement(child) && (child.props as any).children){
                return React.cloneElement(child, {key: i, children: recurseChildren((child.props as any).children)} as any);
            }

            return child;
        });
    };

    return <StyledModal {...props} children={recurseChildren(props.children)} />;
};

Modal.defaultProps = {
    className: '',
    closeOnOverlayClick: true,
    style: {}
};

Modal.displayName = 'Modal';

/**
 * ModalContent
 */
export const ModalContent: React.FC<IModalContentComponentProps> = (props) => {
    const ModalContentComponent = props.isOpen ? StyledModalContentFadeIn : StyledModalContentFadeOut;
    return <ModalContentComponent {...props} />;
};

ModalContent.defaultProps = {
    className: '',
    style: {}
};

ModalContent.displayName = 'ModalContent';

/**
 * ModalHeader
 */
export const ModalHeader: React.FC<IDivBoxComponentProps> = (props) => {
    return <StyledModalHeader boxAs="header" {...props} />;
};

ModalHeader.defaultProps = {
    className: '',
    style: {}
};

ModalHeader.displayName = 'ModalHeader';

/**
 * ModalBody
 */
export const ModalBody: React.FC<IDivBoxComponentProps> = (props) => {
    return <StyledModalBody {...props} />;
};

ModalBody.defaultProps = {
    className: '',
    style: {}
};

ModalBody.displayName = 'ModalBody';

/**
 * ModalFooter
 */
export const ModalFooter: React.FC<IDivBoxComponentProps> = (props) => {
    return <StyledModalFooter boxAs="footer" {...props} />;
};

ModalFooter .defaultProps = {
    className: '',
    style: {}
};

ModalFooter.displayName = 'ModalFooter';

/**
 * ModalCloseButton
 */
export const ModalCloseButton: React.FC<IModalCloseButtonComponentProps> = (props) => {
    const onClick = (e: React.MouseEvent<HTMLDivElement>) => {
        props.onClickButton && props.onClickButton(e);
        props.onClick && props.onClick(e);
    };
    return <StyledModalCloseButton {...props} children="close" onClick={onClick} />;
};

ModalCloseButton .defaultProps = {
    className: '',
    style: {}
};

ModalCloseButton.displayName = 'ModalCloseButton';

/**
 * ModalOverlay
 */
export const ModalOverlay: React.FC<IModalOverlayComponentProps> = (props) => {

    const [show, setShow] = React.useState(props.isOpen);

    React.useEffect(() => {
        if (props.isOpen){
            setShow(true);
        } else {
            setTimeout(() => {
                setShow(false);
            }, 120);
        }
    }, [props.isOpen]);

    const ModalOverlayComponent = props.isOpen ? StyledModalOverlayFadeIn : StyledModalOverlayFadeOut;

    const onClick = (e: React.MouseEvent<HTMLDivElement>) => {
        props.onClickOverlay && props.onClickOverlay(e);
        props.onClick && props.onClick(e);
    };

    if (show){
        return <ModalOverlayComponent {...props} onClick={onClick} />;
    }

    return null;

};

ModalOverlay.defaultProps = {
    className: '',
    style: {}
};

ModalOverlay.displayName = 'ModalOverlay';

export default Modal;
