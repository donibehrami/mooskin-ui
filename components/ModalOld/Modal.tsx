import * as React from 'react';

import styles from './Modal.css';

import { H2 } from '../Headings';
import SmallIconButton from '../SmallIconButton';

export interface IModalProps {
    /** id of the modal */
    id?: string;

    /** modal title */
    title?: string;

    /** override modal styles */
    style?: React.CSSProperties;

    /** override modal class */
    className?: string;

    /** onClick callback function when the background cover is clicked */
    onClickOverlay?: (e: React.MouseEvent<HTMLDivElement>) => void;

    /** onClick callback function when the cloes icon is clicked */
    onClickClose?: (e: React.MouseEvent<HTMLDivElement>) => void;
}

export default class Modal extends React.Component<IModalProps, {}> {
    static defaultProps = {
        className: '',
        style: {},
    };

    static displayName = 'Modal';

    render() {
        return (
            <div className={`modal-component ${styles.container} ${this.props.className}`} id={this.props.id}>
                <div className={styles.modal} style={this.props.style}>
                    {this.getHeader()}
                    {this.props.children}
                </div>
                {this.getOverlay()}
            </div>
        );
    }

    getHeader = () => {
        const closeIcon = <SmallIconButton transparent icon="close" onClick={this.props.onClickClose} />;

        if (this.props.title) {
            return (
                <div className={styles.header}>
                    <H2 style={{ margin: 'auto' }}>{this.props.title}</H2>
                    <div className={styles.iconWrapper}>{closeIcon}</div>
                </div>
            );
        }

        return <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'flex-end' }}>{closeIcon}</div>;
    };

    getOverlay = () => {
        return <div className={styles.cover} onClick={this.props.onClickOverlay} />;
    };
}
