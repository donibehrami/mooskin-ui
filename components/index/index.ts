import '../_utils/globals/global.css';

export { default as Alert, AlertCloseButton, AlertDescription, AlertIcon, AlertTitle } from '../Alert/Alert';
export { default as Skeleton, SkeletonCircle, SkeletonText } from '../Skeleton/Skeleton';
export { default as Drawer, DrawerBody, DrawerCloseButton, DrawerContent, DrawerFooter, DrawerHeader, DrawerOverlay } from '../Drawer/Drawer';
export { default as Stack, HStack, VStack } from '../Stack/Stack';
export { default as Box } from '../Box/Box';
export { default as Carousel } from '../Carousel/Carousel';
export { default as Checkbox, CheckboxDescription, CheckboxIcon, CheckboxLabel } from '../Checkbox/Checkbox';
export { default as DatePicker } from '../DatePicker/DatePicker';
export { default as DateRange } from '../DateRange/DateRange';
export { default as DateSelect } from '../DateSelect/DateSelect';
export { default as Grid, Row, Col } from '../Grid/Grid';
export {
    default as Input,
    InputContainer,
    InputDescription,
    InputEmoji,
    InputIcon,
    InputLabel,
    InputOption,
    InputOptionList,
    InputOptionListTitle,
    InputOverlay
} from '../Input/Input';
export { default as List, ListItem, ListItemBody, ListItemEnd, ListItemHead } from '../List/List';
export { default as Modal, ModalBody, ModalCloseButton, ModalContent, ModalFooter, ModalHeader, ModalOverlay } from '../Modal/Modal';
export { default as Pagination, PaginationButton } from '../Pagination/Pagination';
export { default as Radio, RadioDescription, RadioIcon, RadioLabel } from '../Radio/Radio';
export { default as Sidemenu, SidemenuItem } from '../Sidemenu/Sidemenu';
export { default as IconButton } from '../IconButton/IconButton';
export {
    default as Select,
    SelectContainer,
    SelectDescription,
    SelectFilter,
    SelectIcon,
    SelectLabel,
    SelectLoader,
    SelectOption,
    SelectOptionList,
    SelectOverlay,
    SelectPagination,
    SelectPlaceholder
} from '../Select/Select';
export { default as Switch } from '../Switch/Switch';
export { default as TextArea, TextareaDescription, TextareaLabel } from '../TextArea/TextArea';
export { default as Tabs, TabContent, TabHeader, Tab } from '../Tabs/Tabs';
export { default as Tags, TagClose, TagInput, Tag } from '../Tags/Tags';
export { default as Table, TableHeader, TableHeaderItem, TableRow, TableRowItem } from '../Table/Table';

export { default as IInputCallbackData } from '../_utils';
