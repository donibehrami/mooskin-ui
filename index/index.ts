// Context
export { MooskinContext, MooskinContextProvider, useMooskinContext, withMooskinContext } from '../components/Styled/MooskinContextProvider';

// Components
export { ActionsDropdown, ActionsDropdownArrow, ActionsDropdownItem } from '../components/ActionsDropdown/ActionsDropdown';
export { Alert, AlertCloseButton, AlertDescription, AlertIcon, AlertTitle } from '../components/Alert/Alert';
export { Button } from '../components/Button/Button';
export { Anchor } from '../components/Anchor/Anchor';
export { Skeleton, SkeletonCircle, SkeletonText } from '../components/Skeleton/Skeleton';
export {
    Drawer,
    DrawerBody,
    DrawerCloseButton,
    DrawerContent,
    DrawerFooter,
    DrawerHeader,
    DrawerOverlay
} from '../components/Drawer/Drawer';
export { Stack, HStack, VStack } from '../components/Stack/Stack';
export { Box } from '../components/Box/Box';
export { Carousel } from '../components/Carousel/Carousel';
export { Checkbox, CheckboxDescription, CheckboxIcon, CheckboxLabel } from '../components/Checkbox/Checkbox';
export { DateTimePicker } from '../components/DateTimePicker/DateTimePicker';
export { DatePicker } from '../components/DatePicker/DatePicker';
export { DateRange } from '../components/DateRange/DateRange';
export { DateSelect } from '../components/DateSelect/DateSelect';
export { Description } from '../components/Description/Description';
export { Layout } from '../components/Layout/Layout';
export { Selector, SelectorItem } from '../components/Selector/Selector';
export { Label } from '../components/Label/Label';
export {
    Expandable,
    ExpandableItem,
    ExpandableItemButton,
    ExpandableItemContainer,
    ExpandableItemContent,
    ExpandableItemText
} from '../components/Expandable/Expandable';
export { Grid, Row, Col } from '../components/Grid/Grid';
export {
    Input,
    InputContainer,
    InputEmoji,
    InputIcon,
    InputOption,
    InputOptionList,
    InputOptionListTitle,
    InputOverlay
} from '../components/Input/Input';
export { List, ListItem, ListItemBody, ListItemEnd, ListItemHead } from '../components/List/List';
export { Modal, ModalBody, ModalCloseButton, ModalContent, ModalFooter, ModalHeader, ModalOverlay } from '../components/Modal/Modal';
export { Pagination, PaginationButton } from '../components/Pagination/Pagination';
export { Radio, RadioDescription, RadioIcon, RadioLabel } from '../components/Radio/Radio';
export { Sidemenu, SidemenuItem } from '../components/Sidemenu/Sidemenu';
export { IconButton } from '../components/IconButton/IconButton';
export { Steps, StepContent, StepHeader, Step } from '../components/Steps/Steps';
export {
    Select,
    SelectContainer,
    SelectFilter,
    SelectIcon,
    SelectLoader,
    SelectOption,
    SelectOptionList,
    SelectOverlay,
    SelectPagination,
    SelectPlaceholder
} from '../components/Select/Select';
export { Switch, SwitchHandle } from '../components/Switch/Switch';
export { Loader } from '../components/Loader/Loader';
export { NumberLabel } from '../components/NumberLabel/NumberLabel';
export { LoadingBar } from '../components/LoadingBar/LoadingBar';
export { TextArea } from '../components/TextArea/TextArea';
export { TextEditor } from '../components/TextEditor/TextEditor';
export { Tabs, TabContent, TabHeader, Tab } from '../components/Tabs/Tabs';
export { Tags, TagClose, TagInput, Tag } from '../components/Tags/Tags';
export { Table, TableHeader, TableHeaderItem, TableRow, TableRowItem } from '../components/Table/Table';
export { Footer, FooterBody, FooterEnd, FooterHead } from '../components/Footer/Footer';

// Models
export {
    IActionsDropdownArrowComponentProps,
    IActionsDropdownComponentProps,
    IActionsDropdownItemComponentProps
} from '../components/ActionsDropdown/model';
export { IButtonComponentProps } from '../components/Button/model';
export { IBaseAlertComponentProps } from '../components/Alert/model';
export { IAnchorComponentProps } from '../components/Anchor/model';
export {
    IBaseBoxComponentProps,
    IBoxComponentProps,
    IAnchorBoxComponentProps,
    IBoxLabelComponentProps,
    IButtonBoxComponentProps,
    IInputBoxComponentProps,
    ITextAreaBoxComponentProps
} from '../components/Box/model';
export { ICarouselComponentProps } from '../components/Carousel/model';
export { ICheckboxComponentProps, ICheckboxIconComponentProps } from '../components/Checkbox/model';
export {
    IDatePickerCommonProps,
    IDatePickerComponentProps,
    IDatePickerKeyboardComponentProps,
    PickerType as DatePickerType
} from '../components/DatePicker/model';
export {
    IDateTimePickerCommonProps,
    IDateTimePickerComponentProps,
    IDateTimePickerKeyboardComponentProps,
    PickerType as DateTimePickerType
} from '../components/DateTimePicker/model';
export { IDateRangePickerComponentProps, IRangeSelection } from '../components/DateRange/model';
export { IDateSelectComponentProps } from '../components/DateSelect/model';
export { IDescriptionComponentProps } from '../components/Description/model';
export { ILabelComponentProps } from '../components/Label/model';
export { IDrawerComponentProps, IDrawerContentComponentProps, IDrawerOverlayComponentProps } from '../components/Drawer/model';
export { IExpandableCommonComponentProps, IExpandableComponentProps, IExpandableItemComponentProps } from '../components/Expandable/model';
export { IColProps, IGridProps, IRowProps } from '../components/Grid/model';
export { IIconButtonComponentProps } from '../components/IconButton/model';
export {
    IInputComponentProps,
    IInputContainerComponentProps,
    IInputEmojiComponentProps,
    IInputListComponentProps,
    IInputOptionComponentProps
} from '../components/Input/model';
export { ILayoutComponentProps } from '../components/Layout/model';
export { ILoaderComponentProps } from '../components/Loader/model';
export { ILoadingBarComponentProps } from '../components/LoadingBar/model';
export { IModalComponentProps, IModalContentComponentProps, IModalOverlayComponentProps } from '../components/Modal/model';
export { INumberLabelComponentProps } from '../components/NumberLabel/model';
export { IPaginationButtonComponentProps, IPaginationComponentProps } from '../components/Pagination/model';
export { IRadioComponentProps, IRadioIconComponentProps } from '../components/Radio/model';
export {
    ISelectComponentProps,
    ISelectFilterComponentProps,
    ISelectOptionComponentProps,
    ISelectPaginationComponentProps
} from '../components/Select/model';
export { ISelectorComponentProps, ISelectorItemComponentProps } from '../components/Selector/model';
export { ISidemenuComponentProps, ISidemenuItemComponentProps } from '../components/Sidemenu/model';
export {
    ICommonSkeletonComponentProps,
    ISkeletonCircleComponentProps,
    ISkeletonComponentProps,
    ISkeletonTextComponentProps
} from '../components/Skeleton/model';
export { IStackComponentProps } from '../components/Stack/model';
export { IStepCommonComponentProps, IStepComponentProps, IStepHeaderComponentProps, IStepsComponentProps } from '../components/Steps/model';
export { ISwitchComponentProps, ISwitchHandleComponentProps } from '../components/Switch/model';
export { ITabCommonComponentProps, ITabComponentProps } from '../components/Tabs/model';
export { ITagComponentProps, ITagsComponentProps, ITagsInputComponentProps } from '../components/Tags/model';
export { ITextAreaComponentProps } from '../components/TextArea/model';
export { IPersonalizationTag, ITextEditorComponentProps } from '../components/TextEditor/model';

// Global Styles
export { GlobalStyle } from '../components/Styled/GlobalStyles';

export { default as IInputCallbackData } from '../components/_utils';
