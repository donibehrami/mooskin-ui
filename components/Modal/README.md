# MooSkin UI - Modal Component

___

### Usage

To start using the Modal Component first you have to Import it

```
Import {Modal} from 'mooskin';
```
or modular import
```
// JS
import Modal from 'mooskin/lib/Modal';

// CSS
import 'mooskin/lib/Modal/style.css';
```


And then you can simply start using it by typing

```
<Modal active>
    <whatever/>....
</Modal>
```

Anything can go inside the Modal component.

### Examples

Give it a class 

```
<Modal className={myStyle} />
```

or just give it a custom style

```
<Modal style={yourStyle} />
```

<div class="playground-doc">

## Supported attributes 

* `id` - id of the element
* `active` - wether the modal is active or not.
* `onClickCover` - callback function when the cover behind the modal is clicked, can be used to toggle the modal off.
* `className` - css class
* `style` - Modal style

</div>

Allthough these attributes are supported, all of them are optional.

#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)