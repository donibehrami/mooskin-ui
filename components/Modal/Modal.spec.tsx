import * as React from 'react';
import {Button} from '../index';
import Modal from './Modal';

import { shallow } from 'enzyme';

describe('Modal', () => {

    test('renders correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <Modal
                className="myClass"
                style={{color: 'blue'}}
                id={'button1'}
                onClickCover={func}
                active
            >
                <div>
                    Mooskin
                </div>
            </Modal>
        );
        expect(tree).toMatchSnapshot();
    });

    test('onCoverClick callback is called when cover is clicked', () => {
        const func = jest.fn();

        const component = shallow(<Modal onClickCover={func}>asd</Modal>);

        component.find('.cover').simulate('click');

        expect(func).toHaveBeenCalled();

    });

    test('modal toggles on and off', () => {

        let active = false;

        const on = () => {
            active = false;
        };

        const off = () => {
            active = false;
        };

        const button = shallow(<Button onClick={on}>asd</Button>);
        const component = shallow(<Modal active={active} onClickCover={off}>asd</Modal>);

        expect(component.find('.modalOn')).toBeFalsy;

        button.simulate('click');

        expect(component.find('.modalOn')).toBeTruthy;

    });

});
