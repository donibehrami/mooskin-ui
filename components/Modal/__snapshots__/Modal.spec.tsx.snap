// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<Modal
  className=""
  closeOnOverlayClick={true}
  onClose={[MockFunction]}
  style={Object {}}
>
  <Styled(Box)
    className=""
    closeOnOverlayClick={true}
    onClose={[MockFunction]}
    style={Object {}}
  >
    <Box
      className="sc-gtssRu"
      closeOnOverlayClick={true}
      onClose={[MockFunction]}
      style={Object {}}
    >
      <Component
        className="sc-gtssRu"
        closeOnOverlayClick={true}
        onClose={[MockFunction]}
        style={Object {}}
        theme={
          Object {
            "disabledBackground": "#e2e2e2",
            "disabledfont": "#9d9d9d",
            "inputBackground": "#ffffff",
            "inputBorder": "#e2e2e2",
            "inputFont": "#9d9d9d",
            "label": "#2d2d2d",
            "primary": "#3fbaca",
            "secondary": "#293346",
          }
        }
      >
        <styled.div
          className="sc-gtssRu"
          closeOnOverlayClick={true}
          onClose={[MockFunction]}
          style={Object {}}
          theme={
            Object {
              "disabledBackground": "#e2e2e2",
              "disabledfont": "#9d9d9d",
              "inputBackground": "#ffffff",
              "inputBorder": "#e2e2e2",
              "inputFont": "#9d9d9d",
              "label": "#2d2d2d",
              "primary": "#3fbaca",
              "secondary": "#293346",
            }
          }
        >
          <div
            className="sc-bdnylx dOPKTt sc-gtssRu"
            onClose={[MockFunction]}
            style={Object {}}
          >
            <ModalOverlay
              className=""
              key="0/.0"
              onClick={[Function]}
              style={Object {}}
            />
          </div>
        </styled.div>
      </Component>
    </Box>
  </Styled(Box)>
</Modal>
`;
